src/main/java/org/opentripplanner/routing/algorithm/raptor/transit/cost/DefaultCostCalculator.java

```diff
@@ -111,4 +130,80 @@ public final int calculateMinCost(int minTravelTime, int minNumTransfers) {
             + boardAndTransferCost * minNumTransfers
             + transitFactors.minFactor() * minTravelTime;
     }
+
+    @Override
+    public int costEgress(RaptorTransfer egress) {
+        return egress.hasRides()
+                ? egress.generalizedCost() + transferCostOnly
+                : egress.generalizedCost();
+    }
+
+    /** This is public for test purposes only */
+    public int boardingCostRegularTransfer(
+            boolean firstBoarding,
+            int prevArrivalTime,
+            int boardStop,
+            int boardTime
+    ) {
+        // Calculate the wait-time before the boarding witch should be accounted for in the cost
+        // calculation. Any slack at the end of the last leg is not part of this, because that is
+        // already accounted for. If the previous leg is an access-leg, then it is already
+        // time-shifted, witch is important for this calculation to be correct.
```

Comment:
```suggestion
        // Calculate the wait-time before the boarding which should be accounted for in the cost
        // calculation. Any slack at the end of the last leg is not part of this, because it is
        // already accounted for. If the previous leg is an access leg, then it is already
        // time-shifted, which is important for this calculation to be correct.
```